skrub/_joiner.py

```diff
@@ -166,14 +304,24 @@ def transform(self, X: pd.DataFrame, y=None) -> pd.DataFrame:
         :obj:`~pandas.DataFrame`
             The final joined table.
         """
+        del y
         _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
-        return fuzzy_join(
+        main = self.vectorizer_.transform(
+            X[self._main_key].set_axis(self._aux_key, axis="columns")
+        )
+        match_result = self._matching.match(main, self.max_dist_)
+        aux_table = _join_utils.add_column_name_suffix(self.aux_table, self.suffix)
+        matching_col = match_result["index"].copy()
+        matching_col[~match_result["match_accepted"]] = -1
+        join = pd.merge(
```

Comment:
Using `matching_col` with `right_index` assumes that the auxiliary table has the default index `range(X.shape[0])`, but when this index has been changed to, e.g., `[X.shape[0]-1, ..., 0]`, this assumption doesn't hold anymore.

```python
main_table = pd.DataFrame({"Country": ["France", "Italia", "Spain"]})
aux_table = pd.DataFrame({
    "Country": ["Germany", "France", "Italy"],
    "Capital": ["Berlin", "Paris", "Rome"]
})
aux_table.index = [2, 1, 0]

joiner = Joiner(
    aux_table,
    key="Country",
    suffix="_capitals",
    max_dist=0.9,
    add_match_info=False,
)
joiner.fit_transform(main_table)
#   Country Country_capitals Capital_capitals
# 0  France           France            Paris
# 1  Italia          Germany           Berlin
# 2   Spain              NaN              NaN
```